routesrv/eskipbytes.go

```diff
@@ -112,17 +137,24 @@ func (e *eskipBytes) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
 	e.mu.RLock()
 	count := e.count
 	data := e.data
-	etag := e.etag
+	zdata := e.zdata
+	hash := e.hash
 	lastModified := e.lastModified
 	initialized := e.initialized
 	e.mu.RUnlock()
 
 	if initialized {
-		w.Header().Add("Etag", etag)
-		w.Header().Add("Content-Type", "text/plain; charset=utf-8")
-		w.Header().Add(routing.RoutesCountName, strconv.Itoa(count))
-
-		http.ServeContent(w, r, "", lastModified, bytes.NewReader(data))
+		w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+		w.Header().Set(routing.RoutesCountName, strconv.Itoa(count))
+
+		if strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") && len(zdata) > 0 {
+			w.Header().Set("Etag", `"`+hash+`+gzip"`)
```

Comment:
maybe use fmt.Sprintf?

Reply:
yeah, I though about it but it scatches me to use printf to add quotes

Reply:
```go
w.Header().Set("Etag", `"`+hash+`"`)
w.Header().Set("Etag", `"`+hash+`+gzip"`)
// vs
w.Header().Set("Etag", fmt.Sprintf(`"%s"`, hash))
w.Header().Set("Etag", fmt.Sprintf(`"%s+gzip"`, hash))

```

Reply:
I don't know if `%q` makes it better
```
fmt.Sprintf("%q", hash)
fmt.Sprintf("%q", hash+"+gzip")
```

Reply:
maybe: `fmt.Sprintf("%qgzip", hash)`

Reply:
ah no this does not work https://go.dev/play/p/xy_341u63TE.
I think @AlexanderYastrebov already tried the best one, in case he wants to have a string like:
`"...gzip"` and not `"..."gzip`